plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/ImplementSingleArgumentAggregation.java

```diff
@@ -0,0 +1,237 @@
+        if (outputTypeHandle.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(new JdbcExpression(
+                resultExpression.format(context.getIdentifierQuote().apply(columnHandle.getColumnName()), inputType),
+                outputTypeHandle.get()));
+    }
+
+    public static BuilderExpectingInputType mapAggregation(String functionName)
+    {
+        return builder().setFunctionName(functionName);
+    }
+
+    public static BuilderExpectingInputType mapAggregation(String... functionNames)
+    {
+        return builder().setFunctionNames(functionNames);
+    }
+
+    private static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public interface BuilderExpectingInputType
+            // input type condition is optional
+            extends BuilderExpectingResultType
+    {
+        BuilderExpectingResultType on(Type type);
+
+        BuilderExpectingResultType on(Type... types);
+
+        BuilderExpectingResultType on(Class<? extends Type> type);
+    }
+
+    public interface BuilderExpectingResultType
+            // result type supplier is optional
+            extends BuilderExpectingExpression
+    {
+        default BuilderExpectingExpression returning(Function<Type, Optional<JdbcTypeHandle>> resultTypeHandle)
+        {
+            return returning(Type.class, resultTypeHandle);
+        }
+
+        <T extends Type> BuilderExpectingExpression returning(Class<T> expectedOutputType, Function<T, Optional<JdbcTypeHandle>> resultTypeHandle);
+    }
+
+    public interface BuilderExpectingExpression
+    {
+        AggregateFunctionRule to(String expressionWithPlaceholder);
+    }
+
+    private static class Builder
+            implements BuilderExpectingInputType, BuilderExpectingResultType, BuilderExpectingExpression
+    {
+        private Set<String> functionNames;
+        private Predicate<Type> inputType = type -> true;
+
+        private BiPredicate<Type, Type> inputAndOutputPredicate = Type::equals;
+        private ResultTypeHandleSupplier resultTypeHandle = (outputType, inputType, inputTypeHandle) -> {
+            verify(outputType.equals(inputType));
+            return Optional.of(requireNonNull(inputTypeHandle, "inputTypeHandle is null"));
+        };
+
+        private ResultExpressionFormatter resultExpression;
+
+        private Builder() {}
+
+        public Builder setFunctionName(String functionName)
+        {
+            this.functionNames = ImmutableSet.of(functionName);
+            return this;
+        }
+
+        public Builder setFunctionNames(String... functionNames)
+        {
+            this.functionNames = ImmutableSet.copyOf(functionNames);
+            return this;
+        }
+
+        @Override
+        public Builder on(Type type)
+        {
+            this.inputType = requireNonNull(type, "type is null")::equals;
```

Comment:
IMHO I'd prefer these split into two lines
```suggestion
            requireNonNull(type, "type is null");
            this.inputType = type::equals;
```

Reply:
Usually, I prefer the inlined versions, but here I could go either way.